### PR TITLE
[CI] bundle seed content into Docker image

### PR DESCRIPTION
**Pull request by DelusionalOptimist**
Signed-off-by: Rudraksh Pareek <rudrakshpareek3601@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**
* Docker image now contains patterns and WASM binaries fetched at build time.
* Directory structure:
```
appuser@581632d9aec4:/app/cmd$ ls -R ~/.meshery/seed_content
/home/appuser/.meshery/seed_content:
filters  patterns

/home/appuser/.meshery/seed_content/filters:
binaries

/home/appuser/.meshery/seed_content/filters/binaries:
http_auth_bg.wasm  jwt_filter_bg.wasm  metrics_collector_bg.wasm  singleton_http_call_bg.wasm  singleton_queue_bg.wasm  tcp_metrics_bg.wasm  tcp_packet_parse_bg.wasm

/home/appuser/.meshery/seed_content/patterns:
IstioFilterPattern  bookInfoPattern

```

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -42,6 +42,17 @@ COPY --from=ui /out /app/ui/out
 COPY --from=provider-ui /out /app/provider-ui/out
 COPY --from=wrk2 /wrk2 /app/cmd/wrk2
 COPY --from=wrk2 /wrk2/wrk /usr/local/bin
+
+RUN curl -L -s `curl -s https://api.github.com/repos/DelusionalOptimist/wasm-filters/releases/latest | grep "browser_download_url" | cut -d : -f 2,3 | tr -d '"'` -o wasm-filters.tar.gz \
```
Comment (sayantan1413) [inline Dockerfile:46]:
```suggestion
RUN curl -L -s `curl -s https://api.github.com/repos/layer5io/wasm-filters/releases/latest | grep "browser_download_url" | cut -d : -f 2,3 | tr -d '"'` -o wasm-filters.tar.gz \

```
Should this be layer5io ?

Comment (DelusionalOptimist) [reply Dockerfile:46]: Thanks @sayantan1413.
Yes, you're right this should be. However we might need to think about some things before we publish these binaries at https://github.com/layer5io/wasm-filters.
So this is just a placeholder till then. For demo purposes.

Comment (sayantan1413) [reply Dockerfile:46]: Oh okay got it @DelusionalOptimist .✔️

**Comment by leecalcote** [comment]
@DelusionalOptimist, @Revolyssup, let's acknowledge that the bundled filepath is temporary in that as a subsequent step upon Meshery Server initialization, both registration and copying of the seed content to either the `.meshery/provider/Local` or to `.meshery/provider/<remote>`.

File: Dockerfile [hunk 2]
```diff
@@ -42,6 +42,18 @@ COPY --from=ui /out /app/ui/out
...
+    && tar xzf wasm-filters.tar.gz --directory=/home/appuser/.meshery/seed_content/filters/binaries
+
+RUN curl -L -s https://github.com/service-mesh-patterns/service-mesh-patterns/tarball/master -o service-mesh-patterns.tgz \
+    && mkdir service-mesh-patterns \
+    && mkdir -p /home/appuser/.meshery/seed_content/patterns \
+    && tar xzf service-mesh-patterns.tgz --directory=service-mesh-patterns \
+    && mv service-mesh-patterns/*/samples /home/appuser/.meshery/seed_content/patterns
+
 #COPY --from=nighthawk /nighthawk-go/apinighthawk/bin /usr/local/bin
 RUN curl -Lo nighthawk_service-ubuntu-amd64-v1.0.1.tar.gz https://github.com/layer5io/getnighthawk/releases/download/v1.0.1/nighthawk_service-ubuntu-amd64-v1.0.1.tar.gz \
```
Comment (leecalcote) [inline Dockerfile:58]: @DelusionalOptimist can we avoid pinning to a specific version?

**Comment by leecalcote** [comment]
Running `make docker` with this Dockerfile:

```
 => [stage-4 10/16] COPY --from=wrk2 /wrk2/wrk /usr/local/bin                                                                                                                                              0.0s
 => ERROR [stage-4 11/16] RUN curl -L -s `curl -s https://api.github.com/repos/layer5io/wasm-filters/releases/latest | grep "browser_download_url" | cut -d : -f 2,3 | tr -d '"'` -o wasm-filters.tar.gz   0.5s
------
 > [stage-4 11/16] RUN curl -L -s `curl -s https://api.github.com/repos/layer5io/wasm-filters/releases/latest | grep "browser_download_url" | cut -d : -f 2,3 | tr -d '"'` -o wasm-filters.tar.gz     && mkdir -p /home/appuser/.meshery/seed_content/filters/binaries     && tar xzf wasm-filters.tar.gz --directory=/home/appuser/.meshery/seed_content/filters/binaries:
#35 0.528 curl: no URL specified!
#35 0.528 curl: try 'curl --help' or 'curl --manual' for more information
------
executor failed running [/bin/sh -c curl -L -s `curl -s https://api.github.com/repos/layer5io/wasm-filters/releases/latest | grep "browser_download_url" | cut -d : -f 2,3 | tr -d '"'` -o wasm-filters.tar.gz     && mkdir -p /home/appuser/.meshery/seed_content/filters/binaries     && tar xzf wasm-filters.tar.gz --directory=/home/appuser/.meshery/seed_content/filters/binaries]: exit code: 2
make: *** [docker] Error 1
```

File: Dockerfile (hunk 2 above)
Comment (DelusionalOptimist) [reply Dockerfile:58]: For getnighthawk? Yes we can. However we'll get rid of this and move it out very soon anyways.

**Comment by DelusionalOptimist** [comment]
Yes @leecalcote, that's expected. In my fork I've compiled the filters and published them in a tar in the release: https://github.com/DelusionalOptimist/wasm-filters/releases/tag/v0.1.
However, though you've made the release for upstream wasm-filters repo and changed the path here, that tar is still missing.
I'll create a release workflow for wasm-filters and get those binaries published :+1:

**Review by leecalcote — APPROVED** [review]

**Pull request merged**